File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        int n = nums1.size() + nums2.size();
        int i = 0, j = 0, count = 0;
        
        int idx2 = n / 2;
        int idx1 = idx2 - 1;

        if (nums1.size() == 0) {
            if (nums2.size() % 2 == 1)
                return (double)nums2[nums2.size() / 2];

            else {
                double x = ((nums2[nums2.size() / 2]) +
                            (nums2[(nums2.size() / 2) - 1])) /
                           2.0;
                return x;
            }
        }
        if (nums2.size() == 0) {
            if (nums1.size() % 2 == 1)
                return (double)nums1[nums1.size() / 2];

            else {
                double x = ((nums1[nums1.size() / 2]) +
                            (nums1[(nums1.size() / 2) - 1])) /
                           2.0;
                return x;
            }
        }
cout<<"idx1 "<<idx1<<" idx2 "<<idx2;
        double e1 = 0, e2 = 0;
        while (i < nums1.size() && j < nums2.size()) {

            if (nums1[i] <= nums2[j]) {
                if (count == idx1)
                    e1 = nums1[i];
                if (count == idx2)
                    e2 = nums1[i];
                i++;
                cout << e1 << " i " << e2 << "  " << count << endl;
                count++;
                
            } else {

                if (count == idx1)
                    e1 = nums2[j];
                if (count == idx2)
                    e2 = nums2[j];
                j++;
                count++;
                cout << e1 << " j " << e2 << "  " << count << endl;
            }
        }

        while (i < nums1.size()) {

            if (count == idx1)
                e1 = nums1[i];
            if (count == idx2)
                e2 = nums1[i];
            i++;
            count++;
             cout << e1 << "  " << i << " i1 " << e2 << endl;
        }

        while (j < nums2.size()) {

            if (count == idx1)
                e1 = nums2[j];
            if (count == idx2)
                e2 = nums2[j];
           
             cout << e1 << "  " << count <<"  j= "<<j<< " j1 " << e2 << endl;
            j++;
            count++;
           
        }

        if (n % 2 == 1)
            return e2;
        else
            return (e1 + e2) * 0.5;
    }
};

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
  public double findMedianSortedArrays(int[] arr1, int[] arr2) {
    int n1 = arr1.length, n2 = arr2.length;
    int i = 0, j = 0, k = 0;
    int arr[] = new int[n1+n2];
    while (i < n1 && j < n2) {
      if (arr1[i] <= arr2[j]) {
        arr[k] = arr1[i];
        i++;
        k++;
      } else {
        arr[k] = arr2[j];
        j++;
        k++;
      }
    }

    while (i < n1) {
      arr[k] = arr1[i];
      i++;
      k++;
    }
    while (j < n2) {
      arr[k] = arr2[j];
      j++;
      k++;
    }

    int size = n1 + n2;
    if (size % 2 == 0) {
      return ((arr[size / 2] + arr[size / 2 - 1]) / 2.0);

    }
    return arr[size / 2];
  }
}